src/content/articles/why-typescript-doesnt-follow-strict-semantic-versioning.mdx

```diff
@@ -0,0 +1,300 @@
+
+#### Semver for Developers
+
+Semver acts as a "marketing" number for developers considering whether to update a dependency.
+If you see that a package has, say, only changed its _patch_ version number, you can expect the update will _probably_ not take much effort on your end.
+If, however, the package has updated its _major_ version number, you can expect the update will _probably_ take some manual effort.
+
+:::info
+Regardless of semantic version number changes, it's always a good idea to read a package's release notes before trying to update it.
+You never know when a small change to a dependency will impact your project in meaningful ways.
+:::
+
+#### Semver for Machines
+
+In addition to suggesting change severity to developers, many common tools across the web ecosystem rely on semantic versioning.
+Most notably, [Node.js `package.json`'s package dependencies](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#dependencies) adhere to [`node-semver` versions](https://github.com/npm/node-semver#versions), which directly refer to [semver.org](https://semver.org).
+Symbols such as `^` and `~` commonly seen in `package.json` files are built on semver and explained by [`node-semver`'s documentation](https://github.com/npm/node-semver#advanced-range-syntax).
+
+Another common example of tooling built on semver is automated release management.
+Tools such as [Changesets](https://github.com/changesets/changesets), [release-it](https://github.com/release-it/release-it), and [semantic-release](https://github.com/semantic-release/semantic-release) can automatically publish semantically versioned packages based on the kinds of changes made to the package, such as by [conventional commits](https://www.conventionalcommits.org).
+
+## TypeScript Versioning Today
+
+Given all the advantages of semantic versioning, it can be surprising to learn that TypeScript intentionally does not follow _strict_ semantic versioning.
+TypeScript's version numbers to date have instead incremented according to the following pattern:
+
+- A new _minor_ version of TypeScript (e.g. 4.9.0) is released every three months or so with new features
+  - _Patch_ updates to that version (e.g. 4.9.1) which fix for small bugs are released as needed
+- A new _major_ version of TypeScript (e.g. 5.0.0) is released if the minor version would have exceeded 9 (e.g. 4.10.0)
+
+In other words, TypeScript uses _major_ version numbers more for marketing and public visibility than for semantic versioning.
+This was an intentional engineering decision: releasing a new version of TypeScript almost always requires compiler changes that "break" some existing TypeScript programs.
+
+### "Breaking" Type Checking Changes
+
+Many users consider TypeScript's type checking as a part of its API.
+When a minor or patch version introduces changes to type checkin, some might consider that a "breaking change".
```

Comment:
[Proofreading/question] Is this how checking is supposed to be spelled? 🤔 
```suggestion
When a minor or patch version introduces changes to type checking, some might consider that a "breaking change".
```

Reply:
aha, yeah - fixed locally